prov/shm/src/smr_atomic.c

```diff
@@ -211,11 +203,13 @@ static ssize_t smr_generic_atomic(struct smr_ep *ep,
 	peer_id = smr_peer_data(ep->region)[id].addr.id;
 	peer_smr = smr_peer_region(ep->region, id);
 
-	pthread_spin_lock(&peer_smr->lock);
-	if (peer_smr->cmd_cnt < 2 || smr_peer_data(ep->region)[id].sar_status) {
-		ret = -FI_EAGAIN;
-		goto unlock_region;
-	}
+	if (ofi_atomic_get64(&peer_smr->cmd_cnt) < 2 ||
```

Comment:
What is atomic comparison == 2 supposed to allow?  The checks < 2 look racy to me since we're no longer holding a lock between the check and whatever call is modifying cmd_cnt now.  2 threads could each perform this check, see that cmd_cnt == 2, then both decrement it by 1, leaving no extra commands for the other.  Except that I don't see that cmd_cnt is ever decremented in this patch, only incremented.  Unless I'm missing a decrement call somewhere, I don't see what cmd_cnt does.